src/overlays/actors/ovl_En_Bigpamet/z_en_bigpamet.c

```diff
@@ -90,121 +113,763 @@ static DamageTable D_80A29734 = {
+    if (Math_SmoothStepToF(&this->actor.world.pos.y, this->actor.floorHeight, 0.7f, 30.0f, 10.0f) < 1.0f) {
+        this->unk_29E--;
+        if (this->unk_29E >= 0x10) {
+            func_80A27970(this, play);
+        }
+        if (this->unk_29E == 0) {
+            func_80A28378(this);
+        }
+    } else {
+        func_80A27970(this, play);
+    }
+}
+
+void func_80A28378(EnBigpamet* this) {
+    this->actor.parent->params = 2;
+    this->actor.speedXZ = 0.0f;
+    this->actionFunc = func_80A283A0;
+}
+
+void func_80A283A0(EnBigpamet* this, PlayState* play) {
+    SkelAnime_Update(&this->skelAnime1);
+    if (this->actor.parent->params == 3) {
+        func_80A28E40(this);
+    }
+}
+
+void func_80A283F0(EnBigpamet* this) {
+    Animation_PlayLoop(&this->skelAnime1, &object_tl_Anim_00823C);
+    this->actor.speedXZ = 1.0f;
+    this->actor.world.rot.y = this->actor.shape.rot.y;
+    this->actor.params = 1;
+    this->actionFunc = func_80A2844C;
+}
+
+void func_80A2844C(EnBigpamet* this, PlayState* play) {
+    SkelAnime_Update(&this->skelAnime1);
+
+    if (this->actor.parent->params == 7) {
+        Math_ScaledStepToS(&this->actor.shape.rot.y, Actor_YawBetweenActors(&this->actor, this->actor.parent), 0x400);
+        this->actor.world.rot.y = this->actor.shape.rot.y;
+    } else if (this->actor.parent->params == 8) {
+        this->actor.speedXZ = 0.0f;
+    } else if (this->actor.parent->params == 3) {
+        func_80A28E40(this);
+    }
+}
+
+void func_80A284E4(EnBigpamet* this) {
+    Animation_MorphToPlayOnce(&this->skelAnime1, &object_tl_Anim_001C68, -3.0f);
+    this->unk_29E = 0;
+    this->unk_2A8 = 1.0f;
+    this->unk_2A4 = 1.0f;
+    this->actor.speedXZ = 0.0f;
+    Actor_PlaySfxAtPos(&this->actor, NA_SE_EN_B_PAMET_VOICE);
+    Actor_PlaySfxAtPos(this->actor.parent, NA_SE_EN_FROG_VOICE1);
+    this->actionFunc = func_80A2855C;
+}
+
+void func_80A2855C(EnBigpamet* this, PlayState* play) {
+    if (SkelAnime_Update(&this->skelAnime1)) {
+        func_80A28618(this);
+    } else if (this->skelAnime1.curFrame > 2.0f) {
+        this->unk_2A8 = 1.5f - ((this->skelAnime1.curFrame - 2.0f) * 0.23333333f);
+        this->unk_2A4 = 1.5f - ((this->skelAnime1.curFrame - 2.0f) * 0.083333336f);
+    } else {
+        f32 frame = this->skelAnime1.curFrame;
+        this->unk_2A4 = this->unk_2A8 = (0.25f * frame) + 1.0f;
+    }
+}
+
+void func_80A28618(EnBigpamet* this) {
+    this->actor.draw = func_80A2966C;
+    this->unk_2A8 = 0.5f;
+    this->actor.speedXZ = 0.0f;
+    Actor_PlaySfxAtPos(&this->actor, NA_SE_EN_PAMET_CUTTER_ON);
+    this->actionFunc = func_80A2866C;
+}
+
+void func_80A2866C(EnBigpamet* this, PlayState* play) {
+    this->unk_2A8 += 0.2f;
+    if (this->unk_2A8 > 1.1f) {
+        func_80A286C0(this);
+    }
+}
+
+void func_80A286C0(EnBigpamet* this) {
+    this->collider.info.bumper.dmgFlags = 0xF7CF7FFF;
+    this->collider.base.atFlags |= AT_ON;
+    this->actor.shape.rot.z = 0x680;
+    this->unk_29E = 0xF;
```

Comment:
```suggestion
    this->unk_29E =15;
```
Decremented as timer next function